rclcpp/test/rclcpp/test_publisher.cpp

```diff
@@ -296,6 +296,20 @@ TEST_F(TestPublisher, basic_getters) {
   }
 }
 
+TEST_F(TestPublisher, intra_process_comm_state_publisher) {
+  initialize();
+  // intra-process comm is explicitly disabled
+  rclcpp::PublisherOptionsWithAllocator<std::allocator<void>> options;
+  options.use_intra_process_comm = rclcpp::IntraProcessSetting::Disable;
+  auto publisher = node->create_publisher<test_msgs::msg::Empty>("topic", 10, options);
+  EXPECT_FALSE(publisher->is_intra_process_comm_enabled());
```

Comment:
should it have the scope for each test to make sure the publisher and option is created as we expect? (the same comment goes to subscription.)

```suggestion
  {
    rclcpp::PublisherOptionsWithAllocator<std::allocator<void>> options;
    options.use_intra_process_comm = rclcpp::IntraProcessSetting::Disable;
    auto publisher = node->create_publisher<test_msgs::msg::Empty>("topic", 10, options);
    EXPECT_FALSE(publisher->is_intra_process_comm_enabled());
  }
```

Reply:
Sure that seems to be the way many of the other tests are implemented, just didn't know the extent of that. I'll get that done.